torch/fx/experimental/symbolic_shapes.py

```diff
@@ -164,11 +174,37 @@ def __init__(self, expr, shape_env, constant=None):
         self.constant = constant
 
     def wrap(self, num):
+        if not isinstance(num, float):
+            raise RuntimeError("Wrapping a non-float into a PySymFloat is not valid")
         return PySymFloat(sympy.Float(num), self.shape_env, constant=num)
 
     def __str__(self):
         return f"{self.expr}"
 
+    # You can manually trigger a guard with this function
+    def guard_float(self, file, line):
+        # TODO: use the file/line for some useful diagnostic on why a
+        # guard occurred
+        val = self.shape_env.evaluate_expr(self.expr)
+        try:
+            return float(val)
+        except Exception as e:
+            try:
+                return bool(val)
+            except:
+                # Raise the original error from the float conversion!
+                raise e
+
+    def __sym_int__(self):
+        if SYM_FUNCTION_MODE:
+            return _handle_sym_dispatch(sym_int, (self,), {})
+        # TODO: same considerations as PySymInt.__sym_float__ above
+        return PySymInt(self.expr, self.shape_env)
```

Comment:
This means we can have a PySymInt that wraps a sympy float. Is that ok?
The `__sym_float__` above is already creating `PySymFloat` containing sympy integers so I assume yes?

Reply:
yeah I think this is wrong, we need to truncate in sympy. In my patch, I'm just not finishing the sym int impl.